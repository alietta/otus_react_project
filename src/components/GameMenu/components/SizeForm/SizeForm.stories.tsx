import React from "react";
import { withKnobs, number, array, object } from "@storybook/addon-knobs";
import { SizeForm } from "./SizeForm";
import { DarkMode, LightMode } from "sancho";
import { jsx } from "@emotion/core";

export default {
  title: "SizeForm",
  decorators: [withKnobs],
};

export const darkSizrForm = () => (
  <div css={{ width: 400 }}>
    <DarkMode>
      <SizeForm />
    </DarkMode>
    <LightMode>
      <SizeForm />
    </LightMode>
  </div>
);
