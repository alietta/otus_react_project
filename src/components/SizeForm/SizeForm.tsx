import React, {
  FunctionComponent,
  ChangeEvent,
  useState,
  useEffect,
} from "react";
import { useTheme, InputGroup, Container } from "sancho";

export interface SizeFormProps {
  minWidth: number;
  maxWidth: number;
  minHeight: number;
  maxHeight: number;
  passSize: (values?: { width: number; height: number }) => void;
  label: string;
  default?: { width: number; height: number };
}

const SizeForm: FunctionComponent<SizeFormProps> = ({
  minWidth = 3,
  maxWidth = 10,
  minHeight = 3,
  maxHeight = 10,
  label = "",
  passSize = (): void => {
    console.log("pass size");
  },
}: SizeFormProps) => {
  /* const { minWidth, maxWidth, minHeight, maxHeight, passSize } = props; */
  const theme = useTheme();
  const [size, setSize] = useState<{ width: number; height: number }>({
    width: minWidth,
    height: minHeight,
  });
  const changeSize = (e: ChangeEvent<HTMLInputElement>): void => {
    const newState = { ...size, [e.target.name]: parseInt(e.target.value, 10) };
    setSize(newState);
  };
  useEffect(() => {
    passSize(size);
  }, [size]);

  return (
    <Container
      css={{ background: theme.colors.background.layer, padding: 0, margin: 0 }}
    >
      <form>
        <InputGroup
          label={`${label} width`}
          css={{ flexDirection: "column", display: "flex" }}
        >
          <input
            value={size.width}
            onChange={changeSize}
            type="range"
            min={minWidth}
            max={maxWidth}
            name="width"
            step="5"
          />
        </InputGroup>
        <InputGroup
          label={`${label} height`}
          css={{ flexDirection: "column", display: "flex" }}
        >
          <input
            value={size.height}
            onChange={changeSize}
            type="range"
            min={minHeight}
            max={maxHeight}
            step="5"
            name="height"
          />
        </InputGroup>
      </form>
    </Container>
  );
};

export { SizeForm };
