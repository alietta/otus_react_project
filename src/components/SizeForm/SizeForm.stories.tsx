import React, { FC } from "react";
import { withKnobs } from "@storybook/addon-knobs";
import { SizeForm } from "./SizeForm";
import { DarkMode, LightMode } from "sancho";

export default {
  title: "SizeForm",
  decorators: [withKnobs],
};

export const darkSizeForm: FC = () => {
  const sizeProp = {
    minWidth: 3,
    maxWidth: 10,
    minHeight: 3,
    maxHeight: 10,
    passSize: (): void => {
      console.log("pass size");
    },
  };
  return (
    <div css={{ width: 400 }}>
      <DarkMode>
        <SizeForm {...sizeProp} />
      </DarkMode>
      <LightMode>
        <SizeForm {...sizeProp} />
      </LightMode>
    </div>
  );
};
