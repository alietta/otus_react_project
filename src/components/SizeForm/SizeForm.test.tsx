import React from "react";
import { mount } from "enzyme";
import renderer from "react-test-renderer";
import { SizeForm } from "./SizeForm";

describe("SizeForm", () => {
  it("renders SizeForm", () => {
    expect(renderer.create(<SizeForm />).toJSON()).toMatchSnapshot();
  });
  it("should call passSize with props", () => {
    const passSize = jest.fn();
    const element = mount(<SizeForm passSize={passSize} />);
    element.find("input[name='width']").simulate("change", {
      target: { value: 5, name: "width" },
    });
    element.find("input[name='height']").simulate("change", {
      target: { value: 7, name: "height" },
    });
    expect(passSize).toHaveBeenCalledTimes(3);
    expect(passSize.mock.calls[0][0]).toMatchObject({ width: 3, height: 3 });
    expect(passSize.mock.calls[1][0]).toMatchObject({ width: 5, height: 3 });
    expect(passSize.mock.calls[2][0]).toMatchObject({ width: 5, height: 7 });
  });
});
