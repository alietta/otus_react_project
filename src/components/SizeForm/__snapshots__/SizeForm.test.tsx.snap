// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SizeForm renders SizeForm 1`] = `
.emotion-6 {
  width: 100%;
  max-width: 1200px;
  box-sizing: border-box;
  margin: 0 auto;
  padding: 0 1rem;
  background: white;
}

@media (min-width:992px) {
  .emotion-6 {
    padding: 0 1.5rem;
  }
}

.emotion-2 {
  margin-top: 1rem;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.emotion-2:first-child {
  margin-top: 0;
}

.emotion-1 {
  display: inline-block;
  margin-bottom: 0.5rem;
}

.emotion-0 {
  box-sizing: border-box;
  margin: 0;
  font-weight: 400;
  line-height: 1.5;
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,"Helvetica Neue",Arial,"Noto Sans",sans-serif,"Apple Color Emoji","Segoe UI Emoji","Segoe UI Symbol","Noto Color Emoji";
  font-size: 1rem;
  color: #212529;
  -webkit-font-smoothing: antialiased;
  -webkit-text-size-adjust: none;
  font-size: 0.875rem;
  font-weight: 400;
  color: #5F6871;
  line-height: 1.5;
}

<div
  className="emotion-6"
>
  <form
    style={
      Object {
        "padding": "1.5rem",
      }
    }
  >
    <div
      className="emotion-2"
    >
      <label
        className="Label emotion-1"
        htmlFor=""
      >
        <div
          className="Label__text emotion-0"
        >
          Field width
        </div>
      </label>
      <input
        max={10}
        min={3}
        name="width"
        onChange={[Function]}
        step="1"
        type="range"
        value={3}
      />
    </div>
    <div
      className="emotion-2"
    >
      <label
        className="Label emotion-1"
        htmlFor=""
      >
        <div
          className="Label__text emotion-0"
        >
          Field height
        </div>
      </label>
      <input
        max={10}
        min={3}
        name="height"
        onChange={[Function]}
        step="1"
        type="range"
        value={3}
      />
    </div>
  </form>
</div>
`;
